gobblin-utility/src/main/java/gobblin/util/filesystem/FileStatusEntry.java

```diff
@@ -0,0 +1,211 @@
+package gobblin.util.filesystem;
+
+import java.io.IOException;
+
+import org.apache.commons.io.monitor.FileEntry;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+
+public class FileStatusEntry extends FileStatus {
+
+  static final FileStatusEntry[] EMPTY_ENTRIES = new FileStatusEntry[0];
+
+  private final FileStatusEntry parent;
+  private FileStatusEntry[] children;
+  private final Path path;
```

Comment:
The instance variables in lines 18-23 are unnecessary. They are already part of `FileStatus`.